arch/src/aarch64/fdt.rs

```diff
@@ -81,6 +87,114 @@ pub enum Error {
 }
 type Result<T> = result::Result<T, Error>;
 
+pub enum CacheLevel {
+    /// L1 data cache
+    L1D = 0,
+    /// L1 instruction cache
+    L1I = 1,
+    /// L2 cache
+    L2 = 2,
+    /// L3 cache
+    L3 = 3,
+}
+
+/// NOTE: CACHE SIZE file directory example,
+/// "/sys/devices/system/cpu/cpu0/cache/index0/size".
+pub fn get_cache_size(cache_level: CacheLevel) -> u32 {
+    let mut file_directory: String = "/sys/devices/system/cpu/cpu0/cache".to_string();
+    match cache_level {
+        CacheLevel::L1D => file_directory += "/index0/size",
+        CacheLevel::L1I => file_directory += "/index1/size",
+        CacheLevel::L2 => file_directory += "/index2/size",
+        CacheLevel::L3 => file_directory += "/index3/size",
+    }
+
+    let mut cache_size: u32 = 0;
+    let file_path = Path::new(&file_directory);
+    let file_exist: bool = file_path.exists();
+    if !file_exist {
+        error!("File: {} not exist.", file_directory);
+    } else {
+        info!("File: {} exist.", file_directory);
+
+        let src0 = fs::read_to_string(file_directory).expect("File not exists or file corrupted.");
+        let src = src0.trim();
+        let src_l = &src[0..src.len() - 1];
+        let src_r = &src[src.len() - 1..];
+
+        let size_k: &str = "K";
+        let size_m: &str = "M";
+        let size_g: &str = "G";
+
+        let temp: u32 = src_l.parse().unwrap();
+        if src_r == size_k {
+            cache_size = temp * 1024;
+        } else if src_r == size_m {
+            cache_size = temp * 1024 * 1024;
+        } else if src_r == size_g {
+            cache_size = temp * 1024 * 1024 * 1024;
+        } else {
+            cache_size = temp;
+        }
+    }
+
+    cache_size
+}
+
+/// NOTE: CACHE COHERENCY LINE SIZE file directory example,
+/// "/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size".
+pub fn get_cache_coherency_line_size(cache_level: CacheLevel) -> u32 {
+    let mut file_directory: String = "/sys/devices/system/cpu/cpu0/cache".to_string();
+    match cache_level {
+        CacheLevel::L1D => file_directory += "/index0/coherency_line_size",
+        CacheLevel::L1I => file_directory += "/index1/coherency_line_size",
+        CacheLevel::L2 => file_directory += "/index2/coherency_line_size",
+        CacheLevel::L3 => file_directory += "/index3/coherency_line_size",
+    }
+
+    let mut cache_coherency_line_size: u32 = 0;
+    let file_path = Path::new(&file_directory);
+    let file_exist: bool = file_path.exists();
+    if !file_exist {
+        error!("File: {} not exist.", file_directory);
+    } else {
+        info!("File: {} not exist.", file_directory);
+
+        let src0 = fs::read_to_string(file_directory).expect("File not exists or file corrupted.");
+        let src = src0.trim();
+        cache_coherency_line_size = src.parse().unwrap();
+    }
+
+    cache_coherency_line_size
```

Comment:
Can be simplified a little:
```
    let file_path = Path::new(&file_directory);
    if !file_path.exists() {
        error!("File: {} not exist.", file_directory);
        0
    } else {
        info!("File: {} not exist.", file_directory);

        let src = fs::read_to_string(file_directory).expect("File not exists or file corrupted.");
        src.trim().parse::<u32>().unwrap()
    }
```

Reply:
Thank you very much for your  comment. Your recommendation is perfect, which has improved the quality of the code and made it more elegant. I have refined the code.